src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -135,6 +135,32 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            services.AddSingleton<IAssignmentHandler>(p =>
```

Comment:
Please check the condition before the service registration and let DI create the instance and inject the services instead.

Example: https://github.com/Azure/azure-functions-host/blob/ec3c2d1049351870bbf611760520cf1e48075c8f/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L145-L152